business/iter8.go

```diff
@@ -0,0 +1,279 @@
+package business
+
+import (
+	"encoding/json"
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/models"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	kube "k8s.io/client-go/kubernetes"
+	"github.com/kiali/kiali/config"
+	"k8s.io/client-go/rest"
+	"time"
+	"fmt"
+	"github.com/kiali/kiali/log"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	iter8v1alpha1 "github.com/iter8-tools/iter8-controller/pkg/apis/iter8/v1alpha1"
+	kialiConfig "github.com/kiali/kiali/config"
+)
+
+const GroupName = "iter8.tools"
+const GroupVersion = "v1alpha1"
+
+type Iter8Service struct {
+	k8s kubernetes.IstioClientInterface
+	businessLayer *Layer
+}
+
+func (in *Iter8Service) GetIter8Info() (models.Iter8Info, error) {
+
+	var SchemeGroupVersion = schema.GroupVersion{Group: GroupName, Version: GroupVersion}
+	var iter8Kind = SchemeGroupVersion.WithKind("Experiment")
+
+	if (iter8Kind == schema.GroupVersionKind{}) {
+		return models.Iter8Info{}, nil
+	}
+	log.Infof("Experiment CRD exists")
+	return models.Iter8Info{
+		Enabled: true,
+	}, nil
+}
+
+func (in *Iter8Service) GetIter8Experiment(namespace string, name string) (experiment models.ExperimentDetail, err error) {
+	result := &iter8v1alpha1.Experiment{}
+	cacheToken := ""
+	kConfig := kialiConfig.Get()
+	if kConfig.InCluster {
+		if saToken, err := kubernetes.GetKialiToken(); err != nil {
+			return models.ExperimentDetail{}, err
+		} else {
+			cacheToken = saToken
+		}
+	}
+	config, err := kubernetes.ConfigClient()
+	if err != nil {
+		return models.ExperimentDetail{}, err
+	}
+	istioConfig := rest.Config{
+		Host:            config.Host,
+		TLSClientConfig: config.TLSClientConfig,
+		QPS:             config.QPS,
+		BearerToken:     cacheToken,
+		Burst:           config.Burst,
+	}
+
+	istioClient, err := kubernetes.NewClientFromConfig(&istioConfig)
```

Comment:
I think that in the in *Iter8Service the iter8 REST client can be already created and to avoid to create it on every invocation.

Perhaps, this can be moved to these:
https://github.com/kiali/kiali/blob/master/kubernetes/client.go#L123
https://github.com/kiali/kiali/blob/master/kubernetes/client.go#L313

I know that perhaps this may trigger a couple of refactoring inside.
Let me know what do you think, I can also try to send a PR to your branch with this particular change as I admit that it may need some re-arrange.

Reply:
I will take a look at client.go. There might already the k8s core rest client I can use. Thank you